src/portable/portable.c

```diff
@@ -946,32 +1024,53 @@ static int image_symlink(
 }
 
 static int install_image_symlink(
-                const char *image_path,
+                const Image *image,
+                OrderedHashmap *extension_images,
                 PortableFlags flags,
                 PortableChange **changes,
                 size_t *n_changes) {
 
-        _cleanup_free_ char *sl = NULL;
+        Image *ext;
         int r;
 
-        assert(image_path);
+        assert(image);
 
         /* If the image is outside of the image search also link it into it, so that it can be found with short image
          * names and is listed among the images. */
 
-        if (image_in_search_path(IMAGE_PORTABLE, NULL, image_path))
-                return 0;
+        ORDERED_HASHMAP_FOREACH(ext, extension_images) {
+                _cleanup_free_ char *sl = NULL;
 
-        r = image_symlink(image_path, flags, &sl);
-        if (r < 0)
-                return log_debug_errno(r, "Failed to generate image symlink path: %m");
+                if (image_in_search_path(IMAGE_PORTABLE, NULL, ext->path))
+                        continue;
 
-        (void) mkdir_parents(sl, 0755);
+                r = image_symlink(ext->path, flags, &sl);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to generate image symlink path: %m");
 
-        if (symlink(image_path, sl) < 0)
-                return log_debug_errno(errno, "Failed to link %s %s %s: %m", image_path, special_glyph(SPECIAL_GLYPH_ARROW), sl);
+                (void) mkdir_parents(sl, 0755);
+
+                if (symlink(ext->path, sl) < 0)
+                        return log_debug_errno(errno, "Failed to link %s %s %s: %m", ext->path, special_glyph(SPECIAL_GLYPH_ARROW), sl);
+
+                (void) portable_changes_add(changes, n_changes, PORTABLE_SYMLINK, sl, ext->path);
+        }
+
+        if (!image_in_search_path(IMAGE_PORTABLE, NULL, image->path) && ordered_hashmap_isempty(extension_images)) {
+                _cleanup_free_ char *sl = NULL;
+
+                r = image_symlink(image->path, flags, &sl);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to generate image symlink path: %m");
+
+                (void) mkdir_parents(sl, 0755);
+
+                if (symlink(image->path, sl) < 0)
+                        return log_debug_errno(errno, "Failed to link %s %s %s: %m", image->path, special_glyph(SPECIAL_GLYPH_ARROW), sl);
+
+                (void) portable_changes_add(changes, n_changes, PORTABLE_SYMLINK, sl, image->path);
```

Comment:
instead of repeating basically the same code twice, make this a helper function of its own that you call once for the extensions each and once for the main image from this function?